vim module: allow to set global vimrc

offlinehacker opened the pull request:
###### Motivation for this change

I would like to have global vimrc config. This is optional and should not break anything.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

offlinehacker committed:
nixos/vim: allow to set global vimrc

bjornfor commented:
I don't see the need for this option because it's such a thin wrapper around ```environment.etc."vimrc" = ..."```. But if others like it, then OK.

offlinehacker commented:
Yes, i know, but not everyone knows that you can put global config in `/etc/vimrc`, so from this perspective it kinda makes sense i guess.

Mic92 commented on nixos/modules/programs/vim.nix:
```diff
@@ -12,12 +12,25 @@ in {
     };
   };
 
   config = mkIf cfg.defaultEditor {
```
this means, to set `/etc/vimrc`, vim has to be the default editor?

offlinehacker replied:
Good point, will fix that.

offlinehacker committed:
nixos/vim: add enable option

Adds `enable` option that installs vim regardless if `defaultEditor`
option is set.

offlinehacker commented:
I improved this module a bit, so i added `enable` option that installs vim regardless if it's set as `defaultEditor`. Also `vimrc` can be set regardless if `enable` or `defaultEditor` options are set.

Mic92 commented:
After adding `programs.vim.enable = true;` to my configuration.nix.
I get the following error:

```
sudo nixos-rebuild switch
error: attempt to call something which is not a function but a set, at /home/joerg/git/nixpkgs/nixos/modules/programs/vim.nix:31:12
(use ‘--show-trace’ to show detailed location information)
building Nix...
error: attempt to call something which is not a function but a set, at /home/joerg/git/nixpkgs/nixos/modules/programs/vim.nix:31:12
(use ‘--show-trace’ to show detailed location information)
building the system configuration...
error: attempt to call something which is not a function but a set, at /home/joerg/git/nixpkgs/nixos/modules/programs/vim.nix:31:12
(use ‘--show-trace’ to show detailed location information)
```

teto commented:
I see little advantage if it doesn't replace the (obsolete) vim_configure system. Would you feel like converting it into a module ? You can see in nixos/modules/system/boot/kernel_config.nix how to write a module that can be used independtly of the global config namespace.

mmahut commented:
Are there any updates on this pull request, please?

offlinehacker commented:
I'm closing this, as I do not need it anymore, since I'm using: https://github.com/rycee/home-manager
